### ISSUE TITLE
New home screen

### ISSUE THREAD

**Issue by oldnapalm** [zoffline/zwift-offline#156]
The new home screen introduced in 1.0.100317 can be enabled by changing these 2 variants in https://github.com/zoffline/zwift-offline/blob/3fa696d99d104ef44f28d80d58dc70c1705407b6/zwift_offline.py#L2067

```
                    ('game_1_17_noesis_enabled', 1, None),
                    ('game_1_20_home_screen', 1, None),
```

Thanks Антон Першин (@ursoft)

**Comment by naumovsergey** [comment]
It is not possible to join to events if these  screens are enabled

**Comment by oldnapalm** [comment]
> It is not possible to join to events if these screens are enabled

Are you using the latest version? (post https://github.com/zoffline/zwift-offline/commit/48d3d8ccd6a9c02389083f8b3802249ea8bb1542)

It is possible to join events. Actually it's better than with the default home screen because you join the event world directly, there's no need to switch worlds.

The issue with this new home screen is that you can't join another rider (multiplayer or bot). Not sure if it's designed like that or if we are missing something to display the online players list.

**Comment by naumovsergey** [comment]
> Are you using the latest version? (post [48d3d8c](https://github.com/zoffline/zwift-offline/commit/48d3d8ccd6a9c02389083f8b3802249ea8bb1542))
> 
> It is possible to join events. Actually it's better than with the default home screen because you join the event world directly, there's no need to switch worlds.
> 
> The issue with this new home screen is that you can't join another rider (multiplayer or bot). Not sure if it's designed like that or if we are missing something to display the online players list.

Yes, I did start zoffline server from scratch - the latest zip and new screen enabled . I did have the latest updates of zwift client. There is single player mode. I can see list of event, I can select event but it is not able to select category (A,B,C,D) and not able to join it. 

Also I found that with new home screen it is not possible to select workout with custom route before starting ride - in workout menu after selecting workout zwift offer you four routes. Or you need to start ride than select workout.


May be I did smth wrong.

**Comment by oldnapalm** [comment]
> I can see list of event, I can select event but it is not able to select category (A,B,C,D) and not able to join it.

Odd, not sure what's going on. I can select the category and click "start ride". It says the event will start in 3 hours though, it should be in one minute (I'm in GMT -3). What's your time zone? Maybe the start time is being set in the past.

![2022-01-25 09_23_07-Zwift](https://user-images.githubusercontent.com/38410858/150976836-6b96a941-c872-4e2a-a5ba-59ece3ea0a9b.png)

> Also I found that with new home screen it is not possible to select workout with custom route before starting ride - in workout menu after selecting workout zwift offer you four routes. Or you need to start ride than select workout.

If you click "back" and select another route (even an event), the ride will start with the selected workout.

After starting a ride you can also click "menu", there's a "workouts" button (on the right) where you can select or change the workout.

![2022-01-25 09_24_29-Zwift](https://user-images.githubusercontent.com/38410858/150977087-1051d82f-3af8-4a21-8cdf-c7e70b39a92e.png)

**Comment by naumovsergey** [comment]
> Odd, not sure what's going on. I can select the category and click "start ride". It says the event will start in 3 hours though, it should be in one minute (I'm in GMT -3). What's your time zone? Maybe the start time is being set in the past.

Im in UTC+3 TZ, 
![изображение](https://user-images.githubusercontent.com/61275827/150985616-a73b7ca7-cf86-4c77-879e-790d30dcc806.png)
As you can see I can select e.g. Crit City CW, button is large than other. but nothing happens. does it need to trainer or PWM to be connected?
Old screen does not need it. 


> If you click "back" and select another route (even an event), the ride will start with the selected workout.
Good notice, I did not find it.

**Comment by oldnapalm** [comment]
> As you can see I can select e.g. Crit City CW, button is large than other. but nothing happens. does it need to trainer or PWM to be connected? Old screen does not need it.

No, it doesn't need a paired sensor. For me it says "starts in 3 hr"

![2022_01_25_10_43_53_Zwift](https://user-images.githubusercontent.com/38410858/150988475-299057bb-75aa-4f0a-8b80-b78474e32fce.png)

Try changing this line https://github.com/zoffline/zwift-offline/blob/4f23ad93cd656e45a7d455ae8bd7e5c959a7aafc/zwift_offline.py#L908 to `event_cat.registrationEnd = int(time.mktime(datetime.datetime.now().timetuple())) * 1000 + 60000` or just increase the 60000 to a higher value.

**Comment by oldnapalm** [comment]
> This changes gives me +3hour in old homescreen, but in new homescreen I see nothing - no any upcoming events. If I increase 60000 - the same result - I see changes in old homescreen, but no changes in new one.

Same for me, no events if I use local time. The difference is that for me, if using UTC, events show as starting in 3 hours and I am able to join.

No idea on how to fix. Maybe it's a bug in the client, it's checking event start against local time, although the filter for events listing uses UTC.

**Comment by oldnapalm** [comment]
@naumovsergey

The events issue was (hopefully) fixed in https://github.com/zoffline/zwift-offline/commit/0c0926ac7122e4f458fb9bd29d09286a58e15e17

**Comment by oldnapalm** [comment]
As of version 1.0.100541 in order to enable the new home screen make these changes in `variants.txt`

```
    {
      "name": "game_1_17_noesis_enabled",
      "value": true,
      "values": {}
    },
    {
      "name": "game_1_20_home_screen",
      "value": true,
      "values": {}
    },
```
